src/libraries/System.Runtime.InteropServices/gen/ComInterfaceGenerator/ComInterfaceGenerator.cs

```diff
@@ -66,32 +68,84 @@ public void Initialize(IncrementalGeneratorInitializationContext context)
             var interfacesToGenerate = interfacesWithDiagnostics.Where(static data => data.Diagnostic is null);
             var invalidTypeDiagnostics = interfacesWithDiagnostics.Where(static data => data.Diagnostic is not null);
 
-            var interfaceContexts = interfacesToGenerate.Select((data, ct) =>
+            var interfaceBaseInfo = interfacesToGenerate.Collect().SelectMany((data, ct) =>
```

Comment:
Just a thought, for one pass it probably is better to Collect this and be able to cache the counts, but during incremental passes, this will make us recalculate offsets for all interfaces when 1 is changed, right? Might be premature optimization to change it to something else now, but we could make a note about it for the future.

Reply:
I'd rather do the cache right away as doing this without a cache becomes an `O(n^2)` algorithm almost immediately and heavily derived interface types aren't uncommon (DirectX has cases where they're up to ID3D12Device9 and we have some examples of in dotnet/runtime that go pretty deep like ISOSDacInterface, which is up to ISOSDacInterface13).